tests/components/blebox/test_cover.py

```diff
@@ -0,0 +1,381 @@
+    assert device.model == "shutterBox"
+    assert device.sw_version == "1.23"
+
+
+async def test_init_gatebox(gatebox, hass, config):
+    """Test cover default state."""
+
+    _, entity_id = gatebox
+    entry = await async_setup_entity(hass, config, entity_id)
+    assert entry.unique_id == "BleBox-gateBox-1afe34db9437-position"
+
+    state = hass.states.get(entity_id)
+    assert state.name == "gateBox-position"
+    assert state.attributes[ATTR_DEVICE_CLASS] == DEVICE_CLASS_DOOR
+
+    supported_features = state.attributes[ATTR_SUPPORTED_FEATURES]
+    assert supported_features & SUPPORT_OPEN
+    assert supported_features & SUPPORT_CLOSE
+
+    # Not available during init since requires fetching state to detect
+    assert not supported_features & SUPPORT_STOP
+
+    assert not supported_features & SUPPORT_SET_POSITION
+    assert ATTR_CURRENT_POSITION not in state.attributes
+    assert state.state == STATE_UNKNOWN
+
+    device_registry = await hass.helpers.device_registry.async_get_registry()
+    device = device_registry.async_get(entry.device_id)
+
+    assert device.name == "My gatebox"
+    assert device.identifiers == {("blebox", "abcd0123ef5678")}
+    assert device.manufacturer == "BleBox"
+    assert device.model == "gateBox"
+    assert device.sw_version == "1.23"
+
+
+@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
+async def test_open(wrapper, hass, config):
+    """Test cover opening."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.state = 3  # manually stopped
+
+    def open_gate():
+        feature_mock.state = 1  # opening
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_open = AsyncMock(side_effect=open_gate)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_CLOSED
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_OPEN_COVER, {"entity_id": entity_id}, blocking=True,
+    )
+    assert hass.states.get(entity_id).state == STATE_OPENING
+
+
+@pytest.mark.parametrize("wrapper", ALL_COVER_FIXTURES, indirect=["wrapper"])
+async def test_close(wrapper, hass, config):
+    """Test cover closing."""
+
+    feature_mock, entity_id = wrapper
+
+    def initial_update():
+        feature_mock.state = 4  # open
+
+    def close():
+        feature_mock.state = 0  # closing
+
+    feature_mock.async_update = AsyncMock(side_effect=initial_update)
+    feature_mock.async_close = AsyncMock(side_effect=close)
+
+    await async_setup_entity(hass, config, entity_id)
+    assert hass.states.get(entity_id).state == STATE_OPEN
+
+    feature_mock.async_update = AsyncMock()
+    await hass.services.async_call(
+        "cover", SERVICE_CLOSE_COVER, {"entity_id": entity_id}, blocking=True
+    )
+    assert hass.states.get(entity_id).state == STATE_CLOSING
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)